STX.REST.RESTFulSense.Clients/Services/Foundations/HttpExchanges/HttpExchangeService.Validations.Headers.cs

```diff
@@ -0,0 +1,126 @@
+﻿// ----------------------------------------------------------------------------------
+// Copyright (c) The Standard Organization: A coalition of the Good-Hearted Engineers
+// ----------------------------------------------------------------------------------
+
+using System.Linq;
+using STX.REST.RESTFulSense.Clients.Models.Services.HttpExchanges;
+using STX.REST.RESTFulSense.Clients.Models.Services.HttpExchanges.Exceptions;
+using STX.REST.RESTFulSense.Clients.Models.Services.HttpExchanges.Headers;
+
+namespace STX.REST.RESTFulSense.Clients.Services.Foundations.HttpExchanges
+{
+    internal partial class HttpExchangeService
+    {
+        private static dynamic IsInvalidAcceptHeader(MediaTypeHeader[] mediaTypeHeaders) => new
+        {
+            Condition =
+                mediaTypeHeaders is not null
+                    && (mediaTypeHeaders.Any(header => header is null)
+                        || mediaTypeHeaders.Any(IsInvalidMediaTypeHeader)),
+
+            Message = "Accept header has invalid configuration, fix errors and try again."
+        };
+
+        private static dynamic IsInvalidAcceptCharsetHeader(StringWithQualityHeader[] stringWithQualityHeaders) => new
+        {
+            Condition =
+                stringWithQualityHeaders is not null
+                    && (stringWithQualityHeaders.Any(header => header is null)
+                        || stringWithQualityHeaders.Any(IaInvalidStringWithQualityHeader)),
+
+            Message = "Accept Charset header has invalid configuration, fix errors and try again."
+        };
+
+        private static dynamic IsInvalidAcceptEncodingHeader(StringWithQualityHeader[] stringWithQualityHeaders) => new
+        {
+            Condition =
+                    stringWithQualityHeaders is not null
+                        && (stringWithQualityHeaders.Any(header => header is null)
```

Comment:
why don´t using the same Any for both conditions?

fix naming IaInvalidStringWithQualityHeader